### ISSUE TITLE
Is the correct status code logged?

### ISSUE THREAD

**Issue by gitblit** [cybercoder-naj/logestic#11]
I have a small demo API which I assert with Eden and confirm several of my endpoints return 404s and 400s yet the Logestic logs suggest they are returning 200. Something is not quite right. 🤔

**Comment by cybercoder-naj** [comment]
> I have a small demo API which I assert with Eden and confirm several of my endpoints return 404s and 400s yet the Logestic logs suggest they are returning 200. Something is not quite right. 🤔

May I know which version number you are using and provide some reproducible code?

> I see you are using `afterHandle`. You might consider `OnResponse` or an alternative that someone smarter than me can recommend. You will miss things like validation failures (422) which happen before `beforeHandle`. Validation short-circuits several other life-cycle steps.

I use `onAfterHandle` since I am collecting the status codes and information after the route is handled. Yes, there may be some instances where that may not be a good case. However, if there is a validation error, Elysia redirects you to the `onError` function, which Logestic handles.

**Comment by gitblit** [comment]
I'll see if I can create a minimal repro.

Having said that, I think I have isolated it to this: Logestic is not considering the impact of variations of `return error(400, "")`.

All the cases where I see deviation between my unattractive custom logging with `onResponse` and your library, it is due to this function.

**Comment by cybercoder-naj** [comment]
I've investigated the matter and opened an [issue](https://github.com/elysiajs/elysia/issues/570) on elysia.

**Comment by gitblit** [comment]
Fantastic!

Certainly there are some competing designs in Elysia around this. The docs indicate the lifecycle hooks following `afterHandle` can manipulate the response shape (and possibly the status code). Theoretically, when Elysia hits `onResponse` everything has been resolved to their final values, regardless of competing designs. Is there a reason why `afterHandle` is preferred? The lifecycle [docs suggest](https://elysiajs.com/life-cycle/on-response.html) `onResponse` is better suited for logging and I think Logestic would always be correct if it used that final hook instead of tapping in earlier in the lifecycle.

**Comment by cybercoder-naj** [comment]
I had an extensive review of the lifecycle's page in Elysia, and yes, `onResponse` is a better hook to use for logging. It also better when dealing with `duration` since it gives the total end-to-end duration of the request. 

> I see you switched to onResponse in #12. That feels right to me, but I am curious on your thoughts.

If you notice in this issue thread, I did link this issue to `feat-10/request-timing` since I assumed replacing `onAfterHandle` with `onRequest`. But after testing it, Logestic still loads the incorrect status code. So I've unlinked the PR. I shall investigate this further.

If you find something suspicious in my code, do let me know/open a PR for the fix. Many thanks for your contributions!
